core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -313,6 +319,11 @@ public void close() throws IOException {
     httpClient.close(CloseMode.GRACEFUL);
   }
 
+  @Override
+  public Map<String, String> responseHeaders() {
```

Comment:
I figured this is probably the easiest way to add response headers here. Alternatively, we could add them to `RESTResponse` but that would require more changes.